src/opencl_tezos_fmt_plug.c

```diff
@@ -0,0 +1,488 @@
+	p = strtokm(NULL, "*");
+	cs.raw_address_length = strlen(p) / 2;
+	for (i = 0; i < cs.raw_address_length; i++)
+		cs.raw_address[i] = (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i + 1])];
+
+	MEM_FREE(keeptr);
+	return &cs;
+}
+
+static unsigned int iteration_count(void *salt)
+{
+	struct custom_salt *cs = (struct custom_salt*)salt;
+
+	return cs->iterations;
+}
+
+static void set_salt(void *salt)
+{
+	cur_salt = (struct custom_salt*)salt;
+
+	memcpy(host_salt->pbkdf2.salt, cur_salt->email, cur_salt->email_length);
+	host_salt->pbkdf2.length = cur_salt->email_length;
+	host_salt->pbkdf2.rounds = 2048;
+	memcpy(host_salt->mnemonic, cur_salt->mnemonic, cur_salt->mnemonic_length);
+	host_salt->mnemonic_length = cur_salt->mnemonic_length;
+
+	HANDLE_CLERROR(clEnqueueWriteBuffer(queue[gpu_id], mem_salt,
+		CL_FALSE, 0, sizeof(tezos_salt_t), host_salt, 0, NULL, NULL),
+		"Copy salt to gpu");
+}
+
+static int crypt_all(int *pcount, struct db_salt *salt)
+{
+	const int count = *pcount;
+	size_t gws = count;
+	size_t *lws = (local_work_size && !(gws % local_work_size)) ?
+		&local_work_size : NULL;
+	int i, index;
+	int loops = (host_salt->pbkdf2.rounds + HASH_LOOPS - 1) / HASH_LOOPS;
+
+	// Copy data to gpu
+	BENCH_CLERROR(clEnqueueWriteBuffer(queue[gpu_id], mem_in, CL_FALSE, 0,
+		gws * sizeof(pass_t), host_pass, 0, NULL,
+		multi_profilingEvent[0]), "Copy data to gpu");
+
+	// Run kernel
+	BENCH_CLERROR(clEnqueueNDRangeKernel(queue[gpu_id], crypt_kernel, 1,
+				NULL, &gws, lws, 0, NULL,
+				multi_profilingEvent[1]), "Run kernel");
+
+	for (i = 0; i < (ocl_autotune_running ? 1 : loops); i++) {
+		BENCH_CLERROR(clEnqueueNDRangeKernel(queue[gpu_id],
+					split_kernel, 1, NULL,
+					&gws, lws, 0, NULL,
+					multi_profilingEvent[2]), "Run split kernel");
+		BENCH_CLERROR(clFinish(queue[gpu_id]), "clFinish");
+		opencl_process_event();
+	}
+
+	// Read the result back
+	BENCH_CLERROR(clEnqueueReadBuffer(queue[gpu_id], mem_out, CL_TRUE, 0,
+				gws * sizeof(crack_t), host_crack,
+				0, NULL, multi_profilingEvent[3]), "Copy result back");
+
+	if (!ocl_autotune_running) {
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
+		for (index = 0; index < count; index++) {
+			uint64_t u[8];
+			unsigned char seed[64];
+			unsigned char buffer[20];
+			ed25519_public_key pk;
+			ed25519_secret_key sk;
+			int j;
+
+			memcpy(u, host_crack[index].hash, 64);
+			for (j = 0; j < 8; j++)
+				u[j] = JOHNSWAP64(u[j]);
+			memcpy(seed, u, 64);
+
+			// asymmetric stuff
+			memcpy(sk, seed, 32);
+			ed25519_publickey(sk, pk);
+
+			blake2b((uint8_t *)buffer, (unsigned char*)pk, NULL, 20, 32, 0); // pk is pkh (pubkey hash)
+
+			if (memmem(cur_salt->raw_address, cur_salt->raw_address_length, buffer, 8)) {
+				cracked[index] = 1;
+			} else {
+				cracked[index] = 0;
+			}
+		}
+	}
+
+	return count;
+}
+
+static int cmp_all(void *binary, int count)
+{
+	int index;
+
+	for (index = 0; index < count; index++)
+		if (cracked[index])
+			return 1;
```

Comment:
`any_cracked` would be of even more help here than in the CPU format.

Reply:
I and @magnumripper couldn't get a GPU driven `any_cracked` to work. Maybe, I can simply set `any_cracked` in `crypt_all` by moving this loop there?